content/frameworks/react/virtualize-long-lists-react-window/index.md

```diff
@@ -0,0 +1,286 @@
+
+![Window diagram](./window-diagram.png)
+
+DOM nodes that exit the "window" are recycled, or immediately replaced with newer elements as the user scrolls down the list. This keeps the number of all rendered elements specific to the size of the window.
+
+![DOM nodes changing on scroll](./scroll.gif)
+
+## react-window
+
+**`react-window`** is a third-party library that makes it easier to create virtualized lists in your application. It provides a number of base APIs that can be used for different types of lists and tables.
+
+`react-window` is a rewrite of `react-virtualized`, a different windowing library built by the same author. It's a smaller and faster alternative with APIs and docs that are subjectively easier to use.
+
+<blockquote cite="https://github.com/bvaughn/react-window">
+    <p><i>"Adding a react-virtualized list to a CRA project increases the (gzipped) build size by ~33.5 KB. Adding a react-window list to a CRA project increases the (gzipped) build size by &lt;2 KB."</i></p>
+    <p>
+      <cite>From <a href="https://github.com/bvaughn/react-window">react-window GitHub</a></cite>.
+    </p>
+</blockquote>
+
+### FixedSizeList
+
+Use the **`FixedSizeList`** component if you have a long, one-dimensional list of equally sized items.
+
+```
+import React from 'react';
+import { FixedSizeList } from 'react-window';
+
+const items = [...] // some list of items
+ 
+const Row = ({ index, style }) => (
+  <div style={style}>
+     {/* define the row component using items[index) */}
+  </div>
+);
+ 
+const ListComponent = () => (
+  <FixedSizeList
+    height={500}
+    width={500}
+    itemSize={120}
+    itemCount={items.length}
+  >
+    {Row}
+  </FixedSizeList>
+);
+
+export default ListComponent;
+```
+
+* The `FixedSizeList` component accepts a `height`, `width` and `itemSize` prop to control the size of the items within the list.
+* A function that renders the rows is passed as a child to `FixedSizeList`. Details about the particular item can be accessed with the `index` argument (`items[index]`). 
+* A `style` parameter is also passed in to the row rendering method that **must** be attached to the row element. List items are absolutely positioned with their height and width values assigned inline, and the `style` parameter is responsible for this. 
+
+<div class="aside caution">
+Do not assign <code>height</code> and <code>width</code> properties to the list or the list item with an external CSS file. This would be ignored due to the fact that these style attributes are assigned inline.
+</div>
+
+The Glitch example shown earlier in this article shows an example of a `FixedSizeList` component.
+
+### VariableSizeList
```

Comment:
Similar to title suggestion earlier, retitle this to 'When to use VariableSizeList component'.

Reply:
Changed to `When to use variable size lists`.